dbcrossbarlib/src/clouds/gcloud/bigquery.rs

```diff
@@ -0,0 +1,315 @@
+//! Interfaces to BigQuery.
+
+use serde::de::DeserializeOwned;
+use std::{fs::File, process::Stdio};
+use tempdir::TempDir;
+use tokio::process::Command;
+
+use crate::common::*;
+use crate::drivers::bigquery_shared::{
+    if_exists_to_bq_load_arg, BqColumn, BqTable, TableName,
+};
+use crate::tokio_glue::write_to_stdin;
+
+/// Run a query that should return a small number of records, and return them as
+/// a JSON string.
+async fn query_all_json(ctx: &Context, project: &str, sql: &str) -> Result<String> {
+    // Run our query.
+    debug!(ctx.log(), "running `bq query`");
+    let mut query_child = Command::new("bq")
+        // We'll pass the query on `stdin`.
+        .stdin(Stdio::piped())
+        // We'll read output from `stdout`.
+        .stdout(Stdio::piped())
+        // Run query with no output.
+        .args(&["query", "--headless", "--format=json", "--nouse_legacy_sql"])
```

Comment:
--headless shows up in every command; we could consider factoring it out into an args factory function or similar; but may be overkill; nouse legacy sql also shows up frequently

Reply:
Yeah, we're planning to rip all this out completely and switch to the Rust bindings for Google Cloud.